### PR TITLE
#2777 IEC Try to remove simulation tricks

### PR DESCRIPTION
**Pull request by GerardThevenon**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non-regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated
- [ ] if this commit targets a release branch: the corresponding milestone was added in the ticket and in this PR

### CODE REVIEW

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Sources/IEC/WT4Injector.mo [hunk 1]
```diff
@@ -143,17 +135,14 @@ equation
...
-    Line(points = {{22, -22}, {22, -60}, {-20, -60}, {-20, -74}, {-2, -74}}, color = {0, 0, 127}));
-  connect(add.y, realToComplex1.re) annotation(
-    Line(points = {{22, -80}, {40, -80}, {40, -86}, {58, -86}}, color = {0, 0, 127}));
   connect(elecSystem.uWtImPu, realToComplex1.im) annotation(
     Line(points = {{26, -22}, {26, -74}, {58, -74}}, color = {0, 0, 127}));
   connect(fOCB, genSystem.fOCB) annotation(
     Line(points = {{40, 110}, {40, 60}, {-32, 60}, {-32, 22}}, color = {255, 0, 255}));
   connect(genSystem.PAgPu, PAgPu) annotation(
     Line(points = {{-18, -16}, {0, -16}, {0, -50}, {-80, -50}, {-80, -110}}, color = {0, 0, 127}));
-
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Sources/IEC/WT4Injector.mo:156]: blank line

File: dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Complex/ComplexToPolar.mo [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  Modelica.Blocks.Interfaces.RealOutput len annotation(
+    Placement(transformation(extent={{100,40},{140,80}}), iconTransformation(extent={{100,40}, {140,80}})));
+  Modelica.Blocks.Interfaces.RealOutput phi(unit="rad") annotation(
+    Placement(transformation(extent={{100,-80},{140,-40}}), iconTransformation(extent={{100,-80}, {140,-40}})));
+
+equation
+  if (u.re^2 + u.im^2) > 0 then
+    len = (u.re^2 + u.im^2)^0.5;
+  else
+    len = u.re^2 + u.im^2;
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Complex/ComplexToPolar.mo:32]: Why not simply 0 ?

**Review by cossartque — CHANGES_REQUESTED** [review]

File: dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Complex/ComplexToPolar.mo (hunk 2 above)
Comment (cossartque) [reply dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Complex/ComplexToPolar.mo:32]: And above, why not sqrt ?

**Review by GerardThevenon — COMMENTED** [review]
Approved

**Review by cossartque — APPROVED** [review]

**Pull request merged**